lxd/instance/drivers/driver_qemu.go

```diff
@@ -6579,6 +6579,22 @@ func (d *qemu) MigrateSend(args instance.MigrateSendArgs) error {
 		return fmt.Errorf("Failed receiving migration offer response: %w", err)
 	}
 
+	// If we have unknown fields in the protobuf message, then we received
+	// unexpected data. This will only happen if we were sent a migration.MigrationControl
+	// instead of the expected migration.MigrationHeader. This implies that we have failed on
+	// the target before we were able to send the response header.
+	unknownFields := offerHeader.ProtoReflect().GetUnknown()
```

Comment:
Is this supposed to be `respHeader` instead of `offerHeader`? `offerHeader` is the type being sent, not the one being received for this block.

Reply:
Oops yes, nice catch. Marking as resolved since it is no longer relevant for updated code :)